File: src/com/suushiemaniac/cubing/alglib/modify/mirror/Mirror.java
package com.suushiemaniac.cubing.alglib.modify.mirror;

import com.suushiemaniac.cubing.alglib.modify.Modifier;
import com.suushiemaniac.cubing.alglib.move.Move;

public interface Mirror extends Modifier {
    Move mirror(Move origin);
}

File: src/main/kotlin/com/suushiemaniac/cubing/alglib/transform/mirror/Mirror.kt
package com.suushiemaniac.cubing.alglib.transform.mirror

import com.suushiemaniac.cubing.alglib.transform.Transform
import com.suushiemaniac.cubing.alglib.move.Move

interface Mirror : Transform {
    fun mirror(origin: Move): Move

    override fun transform(origin: Move) = mirror(origin)
}
